### PR TITLE
Near-perfect font selection on Linux

### LINKED ISSUE #1572: registerFont does not allow for two or more fonts with the same (preferred) family name, style, and weight
## Issue or Feature
Cannot use two or more fonts that have the same preferredFamily, and the same style and weight, even though they are different fonts (they do have different family name, different postScriptName).

Example fonts:
1. CopalStd-Solid
 * fontFamily: Copal Std Solid
 * fontSubfamily: Regular
 * preferredFamily: Copal Std
 * preferredSubfamily: Solid
 * postScriptName: CopalStd-Solid
Can download the font here: http://fontsgeek.com/fonts/Copal-Std-Solid

2. CopalStd-Decorated
 * fontFamily: Copal Std Decorated
 * fontSubfamily: Regular
 * preferredFamily: Copal Std
 * preferredSubfamily: Decorated
 * postScriptName: CopalStd-Decorated
 Can download the font here: http://fontsgeek.com/fonts/Copal-Std-Decorated

The above properties can be inspected by uploading the font(s) here: https://opentype.js.org/font-inspector.html and looking at the Naming table.

## Steps to Reproduce

```js
const Canvas = require('canvas')
const fs = require('fs');

Canvas.registerFont(__dirname + '/CopalStd-Decorated.otf', { family: 'Copal Std Decorated'});
Canvas.registerFont(__dirname + '/CopalStd-Solid.otf', { family: 'Copal Std Solid'});

const width = 800, height = 300;
const canvas = Canvas.createCanvas(width, height);
const ctx = canvas.getContext('2d');

ctx.fillStyle = '#FFFFFF';
ctx.fillRect(0, 0, width, height);

ctx.fillStyle = '#FF0000';
ctx.font = '50px Copal Std Solid'
ctx.fillText('Copal Std Solid', 50, 100)
ctx.font = '50px Copal Std Decorated'
ctx.fillText('Copal Std Decorated', 50, 200)

const out = fs.createWriteStream(__dirname + '/test.jpeg')
const stream = canvas.createJPEGStream()
stream.pipe(out)
out.on('finish', () =>  console.log('The JPEG file was created.'))
// etc.
```

The result of this sample code is that both texts are rendered Copal Std Decorated (the font that was registered first), and it's impossible to render a text with Copal Std Solid (font that has the same preferredFamily, even though it is a different font).

test

Expected:
test2

## Your Environment
* Version of node-canvas: 2.6.1
* Environment: node 10.15.3 on Linux

### PR DESCRIPTION
**Pull request by chearon**
Finally done. Removes some GLib because it ended up making that code easier. Also updates the prebuild image so the next prebuilds will have improved matching too.

Fixes #1572

### CODE REVIEW

File: .github/workflows/prebuild.yaml
Comment (zbjornson) [inline .github/workflows/prebuild.yaml:32]: I've been using the `prebuilds` branch to avoid tons of noise on `master`. Just need to remember to change that branch before the next release.

File: src/register_font.cc
Comment (zbjornson) [inline src/register_font.cc:153]: 🥴 I think a leak is possible here from the allocation on 148R, no? It looks like that's not guaranteed to be returned or freed.

Smart pointers or `std::string` would avoid the question.

**Comment by avra-m3** [comment]
Hey @chearon, not sure if this PR is the cause but canvas no longer builds on alpine 
```
make: Entering directory '<project dir>/node_modules/canvas/build'
#12 68.73   SOLINK_MODULE(target) Release/obj.target/canvas-postbuild.node
#12 68.73   COPY Release/canvas-postbuild.node
#12 68.73   CXX(target) Release/obj.target/canvas/src/backend/Backend.o
#12 68.73   CXX(target) Release/obj.target/canvas/src/backend/ImageBackend.o
#12 68.73   CXX(target) Release/obj.target/canvas/src/backend/PdfBackend.o
#12 68.73   CXX(target) Release/obj.target/canvas/src/backend/SvgBackend.o
#12 68.73   CXX(target) Release/obj.target/canvas/src/bmp/BMPParser.o
#12 68.73   CXX(target) Release/obj.target/canvas/src/Backends.o
#12 68.73   CXX(target) Release/obj.target/canvas/src/Canvas.o
#12 68.73   CXX(target) Release/obj.target/canvas/src/CanvasGradient.o
#12 68.73   CXX(target) Release/obj.target/canvas/src/CanvasPattern.o
#12 68.73   CXX(target) Release/obj.target/canvas/src/CanvasRenderingContext2d.o
#12 68.73   CXX(target) Release/obj.target/canvas/src/closure.o
#12 68.73   CXX(target) Release/obj.target/canvas/src/color.o
#12 68.73   CXX(target) Release/obj.target/canvas/src/Image.o
#12 68.73   CXX(target) Release/obj.target/canvas/src/ImageData.o
#12 68.73   CXX(target) Release/obj.target/canvas/src/init.o
#12 68.73   CXX(target) Release/obj.target/canvas/src/register_font.o
#12 68.73 ../src/register_font.cc: In function 'char* get_family_name(FT_Face)':
#12 68.73 ../src/register_font.cc:145:18: error: 'string' is not a member of 'std'
#12 68.73   145 |             std::string best_buf_modified = "@";
#12 68.73       |                  ^~~~~~
#12 68.73 ../src/register_font.cc:21:1: note: 'std::string' is defined in header '<string>'; did you forget to '#include <string>'?
#12 68.73    20 | #include FT_TRUETYPE_IDS_H
#12 68.73   +++ |+#include <string>
#12 68.73    21 | #ifndef FT_SFNT_OS2
#12 68.73 ../src/register_font.cc:146:13: error: 'best_buf_modified' was not declared in this scope
#12 68.73   146 |             best_buf_modified += buf;
#12 68.73       |             ^~~~~~~~~~~~~~~~~
#12 68.73 make: *** [canvas.target.mk:158: Release/obj.target/canvas/src/register_font.o] Error 1
```

**Comment by avra-m3** [comment]
Rolling back the best_buf_modified code to the version you had previously allows you to buiild in alpine, see here https://github.com/avra-m3/node-canvas/commit/e07acd2988eb31b10faa6ead7ae6a38b827b9a7d

File: .github/workflows/prebuild.yaml
Comment (chearon) [reply .github/workflows/prebuild.yaml]: ah, no change needed then. it's already using v9

File: src/register_font.cc
Comment (chearon) [reply src/register_font.cc:154]: That was supposed to use `best_buf` not `buf`, thanks 🤦‍♂️ also switched around the `if`s to do the allocation later.

**Comment by brosenquist** [comment]
@chearon I'm wondering when this can be merged. Looks like the lint is still failing. Is that what is preventing merge?

**Comment by chearon** [comment]
It's just awaiting approval from someone else. Lint is failing on `master` and is fixed by #1986.

**Comment by brosenquist** [comment]
@chearon Thanks for the update. Looks like https://github.com/Automattic/node-canvas/pull/1986 was merged. Yay!  
Sorry if this is a dumb question. I'm not familiar with the process for getting PR approval. Can you tag someone to review/approve or do you just have to wait for someone to decide they want to do it? As you can probably tell I'm really excited to get this fix in. It will solve a bunch of problems for us. Thanks so much for making the fix!

**Review by zbjornson — APPROVED** [review]
Looks good 👍. Just needs the merge conflict fixed (I can do unless you get to it before me).

**Comment by zbjornson** [comment]
Not sure when this will get released unfortunately. I started a branch to try to fix the Windows CI issues (caused by GitHub actions updating to VS2022 by default), but then hit an error that's a pain to troubleshoot through CI logs. Once that's resolved, the same fix will have to be made to the prebuild workflow. 😩 
https://github.com/Automattic/node-canvas/runs/5342435061?check_suite_focus=true

**Comment by brosenquist** [comment]
@zbjornson Thanks again for getting this in. Is the Window CI issue you mentioned related to https://github.com/actions/virtual-environments/issues/4856 or is there some other issue I should be following to keep on top of when a release might happen? I guess I could just wait for a new tag, but I would prefer to know what progress is being made to resolve the issues blocking the release of this fix.

**Comment by zbjornson** [comment]
v2.9.1 is now published.

**Pull request merged**
